pkg/hooks/exec/runtimeconfigfilter.go

```diff
@@ -0,0 +1,33 @@
+package exec
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// RuntimeConfigFilter calls a series of hooks.  But instead of
+// passing container state on their standard input,
+// RuntimeConfigFilter passes the proposed runtime configuration (and
+// reads back a possibly-altered form from their standard output).
+func RuntimeConfigFilter(ctx context.Context, hooks []spec.Hook, config *spec.Spec, postKillTimeout time.Duration) (hookErr, err error) {
+	data, err := json.Marshal(config)
+	for _, hook := range hooks {
+		var stdout bytes.Buffer
+		hookErr, err = Run(ctx, &hook, data, &stdout, nil, postKillTimeout)
+		if err != nil {
+			return hookErr, err
+		}
+
+		data = stdout.Bytes()
+	}
+	err = json.Unmarshal(data, config)
```

Comment:
Could you add a debug log here that the config was overwritten by an OCI hook, to give us bread crumbs to follow in case things go wrong

Reply:
> Could you add a debug log here that the config was overwritten by an OCI hook...

But maybe the hooks returned the config unchanged, in which case logs about hooks possibly having changed the config would just be distracting.  Sounds like you want me to figure out [JSON `interface{}` diffs][1], though?

[1]: https://github.com/containers/libpod/pull/1830#discussion_r234776404

Reply:
We can probably retain the original Config struct and unmarshall into a new struct (this probably will require passing in the generator object so we can reassign it's Config pointer to the new config once we're done). This will let us compare two typed runtime configs, instead of bare bytes or interface{}

Reply:
> ... this probably will require passing in the generator object so we can reassign it's Config pointer to the new config once we're done...

We already pass in a `*Config`, so we can reassign.  So we can add typed comparison without changing the function signature.

> This will let us compare two typed runtime configs, instead of bare bytes or interface{}

Yeah, we can do that if the diff library prefers it.

Reply:
I've [added the logging][1].

[1]: https://github.com/containers/libpod/pull/1830#issuecomment-440055707